### PR TITLE
KTOR-7202 missing respond file overload

### PR DESCRIPTION
**Pull request by jacekgajek**
**Subsystem**
ktor-server-core

**Motivation**
Resolves https://youtrack.jetbrains.com/issue/KTOR-7202/Missing-respondFile-overload

**Solution**
Add missing overload

### CODE REVIEW

**Comment by e5l** [comment]
Hey @jacekgajek, could you please change the target branch to `main`?

**Review by marychatte — APPROVED** [review]
@jacekgajek, thanks for the PR!

File: ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt [hunk 1]
```diff
@@ -9,6 +9,7 @@ import io.ktor.http.content.*
 import io.ktor.server.application.*
 import io.ktor.server.http.content.*
 import java.io.*
+import java.nio.file.Path
```
Comment (osipxd) [inline ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt:12]: nit: `import java.nio.file.*`

File: ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt [hunk 2]
```diff
@@ -52,6 +53,18 @@ public suspend fun ApplicationCall.respondFile(
...
 
+/**
+ * Responds to a client with a contents of a file with the name [fileName] in the [baseDir] folder
+ */
+public suspend fun ApplicationCall.respondFile(
+    baseDir: Path,
+    fileName: Path,
+    configure: OutgoingContent.() -> Unit = {}
+) {
+    val message = LocalFileContent(baseDir, fileName).apply(configure)
```
Comment (osipxd) [inline ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt:64]: I've noticed that `LocalFileContent` is deprecated in favor and `LocalPathContent` should be used instead. Should we also rename `respondFile` into `respondPath`?

File: ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt [hunk 3]
```diff
@@ -60,6 +73,14 @@ public suspend fun ApplicationCall.respondFile(file: File, configure: OutgoingCo
     respond(message)
 }
 
+/**
+ * Responds to a client with a contents of a [file]
+ */
+public suspend fun ApplicationCall.respondFile(file: Path, configure: OutgoingContent.() -> Unit = {}) {
```
Comment (osipxd) [inline ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt:79]: And here `path`, maybe
```suggestion
public suspend fun ApplicationCall.respondFile(path: Path, configure: OutgoingContent.() -> Unit = {}) {
```

**Review by osipxd — COMMENTED** [review]
Good job! Just a couple of stylistic comments

File: ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt [hunk 4]
```diff
@@ -41,14 +42,26 @@ public suspend fun ApplicationCall.respondOutputStream(
 }
 
 /**
- * Responds to a client with a contents of a file with the name [fileName] in the [baseDir] folder
+ * Responds to a client with a contents of a file with the name [relativePath] in the [baseDir] folder
  */
 public suspend fun ApplicationCall.respondFile(
     baseDir: File,
-    fileName: String,
+    relativePath: String,
```
Comment (osipxd) [inline ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt:49]: Couldn't this change cause problems with backward compatibility? In cases when named arguments are used.

File: ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt [hunk 5]
```diff
@@ -41,14 +42,26 @@ public suspend fun ApplicationCall.respondOutputStream(
...
     configure: OutgoingContent.() -> Unit = {}
 ) {
-    val message = LocalFileContent(baseDir, fileName).apply(configure)
+    val message = LocalFileContent(baseDir, relativePath).apply(configure)
+    respond(message)
+}
+
+/**
+ * Responds to a client with a contents of a file with the name [relativePath] in the [baseDir] folder
```
Comment (osipxd) [inline ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt:57]:
```suggestion
 * Responds to a client with a contents of a path designated by [relativePath] in the [baseDir] folder
```

**Review by osipxd — APPROVED** [review]

File: ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt (hunk 4 above)
Comment (marychatte) [reply ktor-server/ktor-server-core/jvm/src/io/ktor/server/response/ApplicationResponseFunctionsJvm.kt:49]: I thought about it, and it seems okay if it goes to the 3.0.0 release, but maybe it's an unneeded change so will revert

**Review by osipxd — APPROVED** [review]

**Pull request merged**